### ISSUE TITLE
NCPA v2.2.1 Services Endpoint issue with search filter

### ISSUE THREAD

**Issue by jvandermeulen** [NagiosEnterprises/ncpa#626]
After updating NCPA from v2.2.0 to v2.2.1 I found a problem with the **services** API endpoint (both on Windows and Linux) that happens with the match type is set to either _search_ or _regex_ and multiple services ought to be returned.

Example:
NCPA version 2.2.0 queries like `-M 'services' -q 'service=app,match=search'`  return both on Windows and Linux several services that match 'app'.

Windows:
`CRITICAL: AppIDSvc is stopped (should be running), AppXSvc is stopped (should be running), AppReadiness is stopped (should be running), AppMgmt is stopped (should be running), COMSysApp is stopped (should be running), Appinfo is running, AppHostSvc is running, TracSrvWrapper is running, Apple Mobile Device Service is running, RpcEptMapper is running
`

Linux:
`OK: systemd-fsck@dev-mapper-datavg\x2ddatalv is running, systemd-fsck@dev-mapper-osvg\x2dtmplv is running, systemd-fsck@dev-mapper-osvg\x2dvarlv is running
`

However version 2.2.1 returns:
```

$ check_ncpa.py -H localhost -t $TOKEN -M services -q 'service=app,match=search'
UNKNOWN: An error occured connecting to API. (HTTP error: '500 INTERNAL SERVER ERROR')
```
When navigating the GUI/API three dots keep rotating on above mentioned search query. The only exception to the '500 INTERNAL SERVER ERROR' seems to be entering a single non-existing  service or entering one existing service.

```
$ check_ncpa.py -H localhost -t $TOKEN -M services -q 'service=mydummyservice,match=search'
UNKNOWN: No services found for service names: mydummyservice
```
Please note the UNKNOWN is expected here.

```
$ check_ncpa.py -H localhost -t $TOKEN -M services -q 'service=crond,match=search'
OK: crond is running
```

**Comment by ssaxnagios** [comment]
I'm able to replicate this as well.

Here are the debug logs:

`2020-04-23 12:25:31,993:DEBUG:handler:Initializing WebSocket
2020-04-23 12:25:31,993:DEBUG:handler:Validating WebSocket request
2020-04-23 12:25:32,069:DEBUG:psapi:Imported windowscounters into the API tree.
2020-04-23 12:25:32,069:DEBUG:psapi:Imported windowslogs into the API tree.
2020-04-23 12:25:32,092:ERROR:app:Exception on /api/services [GET]
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\flask\app.py", line 1817, in wsgi_app
  File "C:\Python27\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
  File "C:\Python27\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
  File "C:\Python27\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
  File "C:\Python27\lib\site-packages\flask\app.py", line 1461, in dispatch_request
  File "C:\ncpa\agent\listener\server.py", line 185, in token_auth_decoration
  File "C:\ncpa\agent\listener\server.py", line 946, in api
  File "C:\ncpa\agent\listener\services.py", line 365, in run_check
ValueError: 'AppIDSvc' is not in list
2020-04-23 12:25:32,094:INFO:handler:127.0.0.1 - - [2020-04-23 12:25:32] "GET /api/services?service=app&match=search&check=true HTTP/1.1" 500 4145 0.101000`

**Comment by jomann09** [comment]
Fixing this in NCPA 2.2.2, this is a duplicate of #620 - I'm a bit busy this week but should be able to get this out later next week if everything goes as planned.

**Comment by ssaxnagios** [comment]
This is still a bug in 2.2.2, adding a +1 for others experiencing it (I can replicate and so can another customer from the ticket below):

https://support.nagios.com/tickets/scp/tickets.php?id=10168

**Comment by jomann09** [comment]
This didn't get updated, 2.2.2 had to go out for some security issues, will get this into the next release.

**Comment by abeerens** [comment]
+1 for experiencing this bug in production.
Please fix this soon.

**Comment by juliusksls** [comment]
Looking forward to 2.2.3 😄 👍 We've skipped 2.2.0 and 2.2.1.

**Comment by yderi** [comment]
+1 for experiencing this bug in production.
Looking forward to 2.2.3

**Comment by jomann09** [comment]
Sorry this was closed prematurely, the fix was not actually in 2.3. It will be in 2.3.1 next week. #742
